//	protoc -I=.\. --java_out=. .\xxxx.proto
//	protoc -I=.\. --cpp_out=. .\xxxx.proto
//package com.appc.tool.proto;
//option java_package = "com.main";

//option optimize_for = SPEED;
//option optimize_for = CODE_SIZE;
option optimize_for = LITE_RUNTIME;
 //	optional 
 
message Resource{
	required string	key = 1;
	required string path = 2; 
	optional string type = 3;
	required int32 size = 4;
	required int32 compresssize = 5;
	optional bool empty = 6;
	optional int32 time = 7;
	optional int32 width = 8;
	optional int32 height = 9;
}

message RPFile{
	required string	name = 1;
	repeated Resource resource = 2;
	required int32 length = 3;
	optional bool ispreload = 4;
}

message PDFile{
	repeated string	name = 1;
}

message ToolConfig{
	repeated string	name = 1;
}

message ServerConfig{
	required string version = 1;
	required string	packagename = 2;
	required string	baseurl = 3;
	repeated RPFile rpfile = 4; // resource package
	optional PDFile pdfile = 5;
	required int32	predownload = 6;
	optional ToolConfig toolconfig =7;
}

